scripts/common_test.py

```diff
@@ -65,9 +66,92 @@
 )
 
 
+class MockCompiler:
+    def wait(self) -> None: # pylint: disable=missing-docstring
+        pass
+
+
+class MockCompilerContextManager():
+    def __init__(self) -> None:
+        pass
+
+    def __enter__(self) -> MockCompiler:
+        return MockCompiler()
+
+    def __exit__(self, *unused_args: str) -> None:
+        pass
+
+
 class CommonTests(test_utils.GenericTestBase):
     """Test the methods which handle common functionalities."""
 
+    def setUp(self) -> None:
+        super().setUp()
+        self.swap_sys_exit = self.swap(sys, 'exit', lambda _: None)
+        def mock_context_manager() -> MockCompilerContextManager:
+            return MockCompilerContextManager()
+        self.swap_ng_build = self.swap(
+            servers, 'managed_ng_build', mock_context_manager)
+        self.print_arr: list[str] = []
+        def mock_print(msg: str) -> None:
+            self.print_arr.append(msg)
+        self.print_swap = self.swap(builtins, 'print', mock_print)
+
+    def test_run_ng_compilation_successfully(self) -> None:
+        swap_isdir = self.swap_with_checks(
+            os.path, 'isdir', lambda _: True, expected_kwargs=[])
+
+        with self.print_swap, self.swap_ng_build, swap_isdir:
+            common.run_ng_compilation()
+
+        self.assertNotIn(
+            'Failed to complete ng build compilation, exiting...',
+            self.print_arr
+        )
+
+    def test_run_ng_compilation_failed(self) -> None:
+        swap_isdir = self.swap_with_checks(
+            os.path, 'isdir', lambda _: False, expected_kwargs=[])
+
+        with self.print_swap, self.swap_ng_build, swap_isdir:
+            with self.swap_sys_exit:
+                common.run_ng_compilation()
+
+        self.assertIn(
+            'Failed to complete ng build compilation, exiting...',
+            self.print_arr
+        )
+
+    def test_subprocess_error_results_in_failed_ng_build(self) -> None:
+        class MockFailedCompiler:
+            def wait(self) -> None: # pylint: disable=missing-docstring
+                raise subprocess.CalledProcessError(
+                    returncode=1, cmd='', output='Subprocess execution failed.')
+
+        class MockFailedCompilerContextManager:
+            def __init__(self) -> None:
+                pass
+
+            def __enter__(self) -> MockFailedCompiler:
+                return MockFailedCompiler()
+
+            def __exit__(self, *unused_args: str) -> None:
+                pass
+
+        def mock_failed_context_manager() -> MockFailedCompilerContextManager:
+            return MockFailedCompilerContextManager()
+
+        self.swap_ng_build = self.swap_with_checks(
+            servers,
+            'managed_ng_build',
+            mock_failed_context_manager
+        )
+        swap_isdir = self.swap_with_checks(os.path, 'isdir', lambda _: False)
```

Comment:
Here you use `self.swap_with_checks` without specifying any expected args/kwargs

Reply:
Done!